### PR TITLE
Reading IID Token from IID Shared Prefs.

### PR DESCRIPTION
**Pull request by ankitaj224**

### CODE REVIEW

File: firebase-installations/src/main/java/com/google/firebase/installations/local/IidStore.java [hunk 1]
```diff
@@ -38,17 +40,79 @@
...
+  }
+
+  private static String getDefaultSenderId(FirebaseApp app) {
+    // Check for an explicit sender id
+    String senderId = app.getOptions().getGcmSenderId();
+    if (senderId != null) {
+      return senderId;
+    }
+    String appId = app.getOptions().getApplicationId();
+    if (!appId.startsWith("1:")) {
```
Comment (andirayo) [inline firebase-installations/src/main/java/com/google/firebase/installations/local/IidStore.java]: There is also version 2: which works exactly the same as 1:

File: firebase-installations/src/main/java/com/google/firebase/installations/local/IidStore.java [hunk 2]
```diff
@@ -38,17 +40,79 @@
...
+      // Not v1, server should be updated to accept the full app ID now
```
Comment (andirayo) [inline firebase-installations/src/main/java/com/google/firebase/installations/local/IidStore.java]: I don't understand the purpose of this comments.
I recommend replacing it with this:
"If applicationId does not contain a (GMP-)App-ID, it contains a Sender identifier"

File: firebase-installations/src/main/java/com/google/firebase/installations/local/IidStore.java [hunk 3]
```diff
@@ -38,17 +40,79 @@
...
+      return appId;
+    }
+    // For v1 app IDs, fall back to parsing the project number out
+    @SuppressWarnings("StringSplitter")
+    String[] parts = appId.split(":");
+    if (parts.length < 2) {
```
Comment (andirayo) [inline firebase-installations/src/main/java/com/google/firebase/installations/local/IidStore.java]: if method #split, splits all parts, then the result should contain exactly 4 parts...

if (parts.length != 4)

File: firebase-installations/src/main/java/com/google/firebase/installations/local/IidStore.java (hunk 1 above)
Comment (andirayo) [reply firebase-installations/src/main/java/com/google/firebase/installations/local/IidStore.java]: if (!appId.startsWith("1:") && !appId.startsWith("2:")) {

Comment (ankitaj224) [reply firebase-installations/src/main/java/com/google/firebase/installations/local/IidStore.java]: Done.

File: firebase-installations/src/main/java/com/google/firebase/installations/local/IidStore.java (hunk 3 above)
Comment (ankitaj224) [reply firebase-installations/src/main/java/com/google/firebase/installations/local/IidStore.java]: Done.

File: firebase-installations/src/main/java/com/google/firebase/installations/local/IidStore.java (hunk 2 above)
Comment (ankitaj224) [reply firebase-installations/src/main/java/com/google/firebase/installations/local/IidStore.java]: Done.

**Pull request merged**